pyproject.toml

```diff
@@ -184,7 +184,7 @@ LD_LIBRARY_PATH = "$(cat /root/.static-deps-path)/lib64:$(cat /root/.static-deps
 
 [tool.cibuildwheel.macos]
 before-build = [
-  "brew install libssh",  # @0.9.4  # pinning the version does not work
+  "MACOSX_DEPLOYMENT_TARGET=10.9 brew install --build-from-source libssh",  # @0.9.4  # pinning the version does not work
```

Comment:
I've identified that `brew` drops env vars like `MACOSX_DEPLOYMENT_TARGET` and they never make it to the underlying compiler.

Could you try pasting instructions from https://github.com/ansible/pylibssh/blob/51f99dd/build-scripts/manylinux-container-image/install_libssh.sh#L74-L90 into this place instead? I don't know if we can reuse this script, which would be good. So I'd start by simply copying stuff.

Technically, you should be able to break into GHA workers via https://github.com/marketplace/actions/debugging-with-tmate to experiment. But bear in mind that those sessions expire in like an hour.